microcosm_pubsub/dispatcher.py

```diff
@@ -107,60 +106,24 @@ def find_handler(self, message, bound_handlers):
         """
         try:
             handler = self.sqs_message_handler_registry.find(message.media_type, bound_handlers)
+            # XXX awkward
```

Comment:
The "logging" function was also wrapping the handler with a decorator that ensures that all loggers have access to `extra` via a function callback. This callback used to call the `sqs_message._context` -- for no good reason since it's always performed inside of the opaque initializer context.

This code needs more testing, but it:
 - Saves the original handler (better abstraction needed)
 - Wraps automatically

Reply:
I've now tested this locally.